src/lib_json/json_value.cpp

```diff
@@ -1018,6 +1018,34 @@ Value Value::removeMember(const std::string& key) {
   return removeMember(key.c_str());
 }
 
+bool Value::removeIndex(ArrayIndex index, Value* removed) {
+  if (this->type_ != arrayValue) {
```

Comment:
I can understand the preference. However, I've taught myself to use `this` always because
 1. It makes the member access obvious.
 2. More important, I have a lot of experience with re-factoring. 

In fact, at my last start-up, because of some brain-dead decisions before I got there, our code took 10-20 minutes to rebuild, in parallel. (Templates can be very slow, and excessive boost::python compounds that problem, wrapping smart-pointers and boost::serialized classes.) To get work done, I would often write free functions first, to avoid altering headers, and I'd move them into the class later. This was particularly effective because most classes were pure-public (again, not my decision).

Just food for thought. It's not a big deal to me either way. Subjectively, I have a slight preference for consistency between internal and external data-member access. But I can see it your way too. And for this code-base, I'm happy to switch it and go with the team.

I'm just glad that you didn't criticize the return-by-pointer. That *is* [a big deal to me](http://stackoverflow.com/questions/26441220/googles-style-guide-about-input-output-parameters-as-pointers).